### ISSUE TITLE
py rtoml library

### ISSUE THREAD

**Issue by abclangs** [HistoricalChristianFaith/Commentaries-Database#7]
Could you please share that library with us?

**Comment by emeth-** [comment]
Absolutely, the library is here:
https://pypi.org/project/rtoml/

As the Readme notes:
>  It's a python3 script, and requires you to "pip install rtoml" first

If you need to install pip, here's some details on how to do so:
https://pip.pypa.io/en/stable/installation/

**Comment by abclangs** [comment]
Thanks man. How often do u add new comments? It would be great if u add more early christian comments

**Comment by emeth-** [comment]
Checkout the commits to see the history: https://github.com/HistoricalChristianFaith/Commentaries-Database/commits/master

I tend to add new ones / update existing ones every few days, with some breaks for real-life things that pop up.

**Comment by abclangs** [comment]
Thanks. I have a question. Is every antenicene and nicene commentary in there? And what do u mean by "update existing ones"?

**Comment by emeth-** [comment]
> Is every antenicene and nicene commentary in there?

Not yet!

> And what do u mean by "update existing ones"?

There's two ways commentaries get added. The first is when someone (so far, just me) does a deep dive on a particular verse, gathering every historical church father commentary/interpretation of that verse that is extant and uploading them to this database. I'm currently working on doing so right now for Matthew 5:31-32.

The second is when a bulk automated data load is done from an external source - recently I did that for all of Augustine's commentaries on the Psalms (which was a bit tricky, as Augustine used the Septuagint numbering system for Psalms, but this repository uses the Masoretic numbering).

The second method sometimes has errors. Sometimes it's formatting errors, sometimes the commentary is cut-off in the wrong spot, and sometimes the commentary is even misattributed. 

For example, months ago I did a bulk data load of commentaries by Clement of Rome - but they lacked attribution (e.g. didn't have the name of the work that the quote was from). A few weeks ago I went through all of those commentaries by hand and dug into the original source for each one so I could add proper attribution / the name of Clement of Rome's work in which he said that quote. In doing so, I found a handful of those commentaries attributed to Clement of Rome were actually from Clement of Alexandria - so I fixed that.

**Comment by luisgutie** [comment]
Great man. I really appreciate your work. What external source do u use to make a bulk load? I would like to help you

**Comment by abclangs** [comment]
Great man. I really appreciate your work. What external source do u use to make a bulk load? I would like to help you. And what percentage of error do u think this commentaries database has? What is your plan for lxx vs masoretic numbering?

**Comment by emeth-** [comment]
Any help is always appreciated. I've bulk loaded all extant databases I'm aware of - the main issue with those databases is that they simply have the commentary/quote without attribution, and often had a computer determine where to start/end the quote. It's a lot of manual labor that I hope to be shared (thus this open database) to review all those, add attribution, and update the quote if the start/end of the quote is in the middle of a sentence or something.

In addition to that, I occassionally write scripts to bulk load directly from church father commentaries not already in the database (I just find that church father's commentary published online in english in the public domain, and parse it with a python script. That's what I did for Augustine's commentaries on the psalms.

Over in the Writings Database repository (https://github.com/HistoricalChristianFaith/Writings-Database), you'll find various texts that were not in the public domain in English that I've been porting over, using ChatGPT / Google Translate to translate them. I also load those in, as I get them translated.

For LXX vs Masoretic numbering, we've simply adopted  the Masoretic numbering here. That seems to be the more common one in English-speaking countries, and we're compiling the commentaries in English - so it just made sense.
